tests/test_indexer.py

```diff
@@ -105,7 +105,7 @@ def tearDownClass(cls):
 
     def setUp(self):
         super().setUp()
-        self.indexer = self.indexer_cls()
+        self.indexer = self.indexer_cls(backend.ElasticsearchIndexBackend())
```

Comment:
Part of the benefit of the separation of the indexing logic and the actual storage implementation is that this could be replaced with an in-memory implementation, no? z.B. have a `class InMemoryIndexBackend(IndexBackend)`. This might finally make this test fast!

If possible, I guess this improvement could go in a different commit.